src/briefcase/integrations/java.py

```diff
@@ -11,11 +12,11 @@
 
 
 class JDK:
-    name = "java"
+    name = "jdk"
```

Comment:
So, these changes overload the uses of the `name` attribute for tools by requiring `name` to match the name of the attribute used on a `ToolCache` for the tool. The driving factor behind this was allowing `UpgradeCommand` a way to find the tool after verification puts the tool in the cache.

The impact to end-users from this change is `briefcase upgrade java` no longer works since they would need to use `briefcase upgrade jdk`.

To avoid this, I would instead need to change the `tools.jdk` refernce to `tools.java`. I'm not sure if there's a preference here or a concern that users have actually scripted `briefcase upgrade java` somewhere.

Reply:
Yeah - this is a weird one. End users will almost certainly be thinking of this as "java", but what they're actually using is a JDK. The configuration variables describe "JAVA_HOME"; and the install folder is `~briefcase/tools/java`; but the download is `jdk-X.tar.gz`. And, after all that... JDK only needs to exist as a dependency of Android, so it's not actually being used by end users anyway.

Purely cosmetically, I think my preference would be to call the tool `tools.java` - but that's a mild preference. If that's too much churn, I'm not going to get too bent out of shape if we need to call it `tools.jdk`.

Reply:
Renamed `tools.jdk` to `tools.java`.